testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -4520,22 +4520,87 @@ void testBitGetFunc(SqlOperatorFixture f, String functionName) {
     f.checkNull("to_hex(cast(null as varbinary))");
   }
 
+  @Test void testHex() {
```

Comment:
Could be helpful to leave a comment to link jira case.
For example 
```
/** Test case for
 * <a href="https://issues.apache.org/jira/browse/CALCITE-4861">[CALCITE-4861]
 * Optimization of chained CAST calls leads to unexpected behavior</a>. */
```

Reply:
Resolved.